src/openrc/rc.c

```diff
@@ -528,6 +528,7 @@ runlevel_config(const char *service, const char *level)
 	bool retval;
 
 	dir = dirname(init);
+	dir = dirname(dir);
```

Comment:
I suggest putting a comment explaining what you've said in the commit message.

Reply:
Personally, I think this is fine. But maybe you can just skip the first assignment and chain the calls:

```c
dir = dirname(dirname(init));
```